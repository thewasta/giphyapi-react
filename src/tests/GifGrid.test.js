import React from "react";
import GifGrid from "../components/GifGrid";
import {shallow} from "enzyme";
import {useFetchGifs} from "../hooks/useFetchGifs";

jest.mock("../hooks/useFetchGifs");
describe("Pruebas de GifGrid", () => {
    const category = "Goku";
    test("should show the correct component and match with the snapshot", () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de mostrar items cuando se cargar im√°genes con useFetchGif", function (assert) {
        const gifs = [{
            id: "ABC",
            url: "https://cualqircosa.com/abc.jpg",
            title: "Cualquier cosa"
        }];
        useFetchGifs.mockReturnValue({
            data: [gifs],
            loading: true
        });
        const wrapper = shallow(<GifGrid category="Dragon"/>);
        expect(wrapper).toMatchSnapshot();
        assert();
    });
});