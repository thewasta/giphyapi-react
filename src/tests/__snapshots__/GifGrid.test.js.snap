// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid debe de mostrar items cuando se cargar im√°genes con useFetchGif 1`] = `
<Fragment>
  <h3>
    Dragon
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid "
  >
    <GifGridItem
      0={
        Object {
          "id": "ABC",
          "title": "Cualquier cosa",
          "url": "https://cualqircosa.com/abc.jpg",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`Pruebas de GifGrid should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;
